### PR TITLE
[lldb][test] Remove `self` references from decorators

### PR DESCRIPTION
**Pull request by rupprecht**
This is partial step toward removing the vendored `unittest2` dep in favor of the `unittest` library in standard python. One of the large differences is when xfail decorators are evaluated. With the `unittest2` vendored dep, this can happen at the moment of calling the test case, and with LLDB's decorator wrappers, we are passed the test class in the decorator arg. With the `unittest` framework, this is determined much earlier; we cannot decide when the test is about to start that we need to xfail.

Fortunately, almost none of these checks require any state that can't be determined statically. For this patch, I moved the impl for all the checks to `lldbplatformutil` and pointed the decorators to that, removing as many `self` (i.e. test class object) references as possible. I left wrappers within `TestBase` that forward to `lldbplatformutil` for convenience, but we should probably remove those later.

The remaining check that can't be moved statically is the check for the debug info type (e.g. to xfail only for dwarf). Fixing that requires a different approach, so I will postpone that to the next patch.

### CODE REVIEW

**Review by JDevlieghere — APPROVED** [review]
👏 Thanks for doing this!

File: lldb/packages/Python/lldbsuite/test/lldbplatformutil.py [hunk 1]
```diff
@@ -184,3 +187,144 @@ def hasChattyStderr(test_case):
...
+        arch = "x86_64"
+    if arch in ["armv7l", "armv8l"]:
+        arch = "arm"
+    return arch
+
+
+lldbArchitecture = None
+
+
+def getLldbArchitecture():
```
Comment (medismailben) [inline lldb/packages/Python/lldbsuite/test/lldbplatformutil.py:210]: I think we can make an exception for lldb's casing here.

```suggestion
def getLLDBArchitecture():
```

File: lldb/packages/Python/lldbsuite/test/lldbtest.py [hunk 2]
```diff
@@ -1310,82 +1308,29 @@ def isAArch64Windows(self):
...
-        module = builder_module()
-        arch = module.getArchitecture()
-        if arch == "amd64":
-            arch = "x86_64"
-        if arch in ["armv7l", "armv8l"]:
-            arch = "arm"
-        return arch
+        return lldbplatformutil.getArchitecture()
 
     def getLldbArchitecture(self):
```
Comment (medismailben) [inline lldb/packages/Python/lldbsuite/test/lldbtest.py:1313]:
```suggestion
    def getLLDBArchitecture(self):
```

File: lldb/packages/Python/lldbsuite/test/lldbtest.py [hunk 3]
```diff
@@ -1310,82 +1308,29 @@ def isAArch64Windows(self):
...
-            str = output.decode()
-
-            for line in str.splitlines():
-                m = re.search(r"Current executable set to '.*' \((.*)\)\.", line)
-                if m:
-                    self.lldbArchitecture = m.group(1)
-                    break
-
-        return self.lldbArchitecture
+        return lldbplatformutil.getLldbArchitecture()
```
Comment (medismailben) [inline lldb/packages/Python/lldbsuite/test/lldbtest.py:1315]:
```suggestion
        return lldbplatformutil.getLLDBArchitecture()
```

**Comment by medismailben** [comment]
Nice! LGTM with some nits.

**Review by medismailben — APPROVED** [review]

**Review by bulbazord — APPROVED** [review]
Beautiful, thank you for working on this.
After this change, what's left to get us using the python-provided `unittest`?

File: lldb/packages/Python/lldbsuite/test/lldbplatformutil.py (hunk 1 above)
Comment (rupprecht) [reply lldb/packages/Python/lldbsuite/test/lldbplatformutil.py]: Thanks, applied

**Comment by rupprecht** [comment]
> Beautiful, thank you for working on this. After this change, what's left to get us using the python-provided `unittest`?

The giant lambda inside `_decorateTest` has one remaining `self` reference, which is `self.getDebugInfo()` -- i.e. the debug info specific variant created by the metaclass factory. My current approach is basically to annotate the test in a similar way as we do for the `@no_debug_info_test`, with some state that indicates which debug info kinds should be skipped/xfailed, and then when the metaclass creates each variant, it can create them as being wrapped w/ `@skip` or `@expectedFailure` depending on the result of that.

After that, not a lot. Both @JDevlieghere and I have made an attempt at this in the past, and so we've already done a lot of cleanup to make things work with either library.

**Pull request merged**